service/storage_service.cc

```diff
@@ -4412,7 +4412,62 @@ future<> storage_service::move(token new_token) {
 
 future<std::vector<storage_service::token_range_endpoints>>
 storage_service::describe_ring(const sstring& keyspace, bool include_only_local_dc) const {
-    return locator::describe_ring(_db.local(), _gossiper, keyspace, include_only_local_dc);
+    size_t nr_tablet_tables = 0;
+    co_await _db.local().get_tables_metadata().for_each_table_gently(coroutine::lambda([&] (table_id id, lw_shared_ptr<replica::table> table) -> future<> {
+        auto s = table->schema();
+        if (s->ks_name() == keyspace && table->uses_tablets()) {
+            ++nr_tablet_tables;
+        }
+        co_return;
+    }));
+    if (nr_tablet_tables) {
```

Comment:
A keyspace either has only tablet tables or only vnode tables. You don't even need to check the tables to find this out, you can check `_db.local().find_keyspace(keyspace).get_abstract_replication_strategy().uses_tablets()`.

Reply:
I looked for the keyspace level check and did not find one. Thanks, this is exactly what I wanted.

Reply:
One day this won't be true, when we have in-place migration from vnodes to tablets.